### PR TITLE
Added templating support in web hook security rules

### PR DESCRIPTION
**Pull request by sharadregoti**

### CODE REVIEW

File: gateway/modules/auth/match.go [hunk 1]
```diff
@@ -72,8 +74,48 @@ func (m *Module) matchRule(ctx context.Context, project string, rule *config.Rul
...
+			t := template.New(rule.Name)
+			t = t.Funcs(tmpl2.CreateGoFuncMaps(m))
+			t, err = t.Parse(rule.ReqTmpl)
+			if err != nil {
+				return formatError(ctx, rule, helpers.Logger.LogError(helpers.GetRequestID(ctx), "Unable to parse provided template in security rule (Webhook)", err, nil))
+			}
+			if rule.OpFormat == "" {
+				rule.OpFormat = "json"
+			}
+			obj, err = tmpl2.GoTemplate(ctx, t, rule.OpFormat, token, claims, args["args"])
```
Comment (YourTechBud) [inline gateway/modules/auth/match.go:108]: put the original token and claims here

File: gateway/utils/tmpl/go.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ import (
 )
 
 // GoTemplate executes a go template
-func GoTemplate(module, segment string, tmpl *template.Template, format, token string, claims, params interface{}) (interface{}, error) {
+func GoTemplate(ctx context.Context, tmpl *template.Template, format, token string, claims, params interface{}) (interface{}, error) {
```
Comment (YourTechBud) [inline gateway/utils/tmpl/go.go:19]: You took a context but are not using it. I suggest you use it instead of `context.Todo`

**Review by YourTechBud — CHANGES_REQUESTED** [review]

File: gateway/modules/auth/match.go (hunk 1 above)
Comment (sharadregoti) [reply gateway/modules/auth/match.go:108]: If I pass original token here, then what is the use of creating new token & claims if we are not using it?

Comment (YourTechBud) [reply gateway/modules/auth/match.go:108]: The point of claims is to generate a custom token to override the original token in the Authorisation header. This is useful when the webhook is on a SC endpoint and we want to make sure the request was made internally.

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**